DNN Platform/Library/Entities/Portals/PortalSettingsService.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.Entities.Portals
+{
+    using System;
+    using System.Collections.Generic;
+
+    using DotNetNuke.Abstractions.Settings;
+    using DotNetNuke.Common.Utilities;
+    using DotNetNuke.Entities.Host;
+    using DotNetNuke.Instrumentation;
+    using DotNetNuke.Security;
+
+    /// <summary>
+    /// Provides an implementation of the <see cref="ISettingsService"/>
+    /// used for updating the Portal Settings. This is instantiated by the
+    /// <see cref="PortalSettingsManager"/>.
+    /// </summary>
+    public class PortalSettingsService : ISettingsService
+    {
+        private static readonly ILog Logger = LoggerSource.Instance.GetLogger(typeof(PortalSettingsService));
+
+        /// <summary>
+        /// The callback for retrieving the most up to date settings data.
+        /// </summary>
+        protected Func<IDictionary<string, string>> LoadSettingsCallback { get; }
+
+        /// <summary>
+        /// Gets the current Portal Settings.
+        /// </summary>
+        protected IDictionary<string, string> Settings { get; private set; }
+
+        /// <summary>
+        /// Gets the save service, used for saving the
+        /// current portal settings.
+        /// </summary>
+        protected ISaveSettingsService SaveService { get; }
+
+        /// <summary>
+        /// Gets the delete service, used for deleting
+        /// a portal setting.
+        /// </summary>
+        protected IDeleteSettingsService DeleteService { get; }
+
+        /// <inheritdoc />
+        public string this[string key]
+        {
+            get => this.Get(key);
+            set => this.Update(key, value);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PortalSettingsService"/> class.
+        /// </summary>
+        /// <param name="loadSettingsCallback">A callback to retrieve the portal settings.</param>
+        /// <param name="saveService">The save settings implementation.</param>
+        /// <param name="deleteService">The delete settings implementation</param>
+        public PortalSettingsService(Func<IDictionary<string, string>> loadSettingsCallback, ISaveSettingsService saveService, IDeleteSettingsService deleteService)
+        {
+            this.LoadSettingsCallback = loadSettingsCallback;
+            this.SaveService = saveService;
+        }
+
+        /// <inheritdoc />
+        public T Get<T>(string key)
+            where T : IConvertible
+        {
+            try
+            {
+                // DNN uses special logic for boolean. It accepts Y and TRUE. We should make sure that is convertible
+                return (T)Convert.ChangeType(this[key], typeof(T));
```

Comment:
This is where I would like to see the logic from `SerializationController` reused (which will handle the special Boolean logic, for example).